File: lib/mediaplayer/media_options.dart
import 'package:flutter/cupertino.dart';

class MediaOptions {
  // TODO Map these to shared preferences (flutter equivalent).
  // TODO Save when changed and load at runtime.

  // Misc Options
  static String audioGroupBy = "album";

  // Times
  static int seekForwardTime = 7500;
  static int seekBackwardsTime = 3500;

  // Sizes
  static const double iconsize = 48.0;
  static double videoPlayBtnMultiplier = 2.25;
  static double playBtnMultiplier = videoPlayBtnMultiplier / 1.5;
  static const fabSize = 48.0;

  // Colors
  static const Color selectedColor = Color.fromARGB(152, 82, 255, 31);
  static const Color controlGroupBackgroundColor = Color.fromARGB(165, 145, 145, 145);
  static const Color fabBackgroundColor = Color.fromARGB(215, 195, 195, 195);
  static const Color superMediaButtonColor = Color.fromARGB(205, 235, 235, 235);
  static const Color dragHighlightColor = Color.fromARGB(139, 255, 157, 0);
  static const Color speedSelectorBackground = Color.fromARGB(198, 56, 234, 169);
}

File: android/app/src/main/kotlin/com/stephenmorgandevelopment/super_media_bros_3/preferences/MediaOptions.kt
package com.stephenmorgandevelopment.super_media_bros_3.preferences

import android.graphics.Color
import com.stephenmorgandevelopment.super_media_bros_3.models.Audio
import com.stephenmorgandevelopment.super_media_bros_3.models.MediaGroup


// TODO Pull values from shared preferences.
// TODO These will either be saved from the flutter side, or passed here to be
// TODO saved from the MediaOptions class in flutter.
object MediaOptions {

    // Misc Options
    val defaultGroupBy = Audio.Category.ALBUMS

    // Times
    val seekForwardTime = 7500
    val seekBackwardsTime = 3500

    // Sizes
    val  iconsize = 48.0
    val  videoPlayBtnMultiplier = 2.25
    val playBtnMultiplier = videoPlayBtnMultiplier / 1.5
    val fabSize = 48.0

    // Colors
    val  selectedColor = Color.argb(152, 82, 255, 31);
    val  controlGroupBackgroundColor = Color.argb(165, 145, 145, 145)
    val fabBackgroundColor = Color.argb(215, 195, 195, 195)
    val superMediaButtonColor = Color.argb(205, 235, 235, 235)
    val dragHighlightColor = Color.argb(139, 255, 157, 0)
    val speedSelectorBackground = Color.argb(198, 56, 234, 169)
}